beetmoverscript/src/beetmoverscript/script.py

```diff
@@ -285,6 +290,208 @@ async def upload_data(context):
         await retry_data_upload(context, dataUpload["destinations"], data, dataUpload["contentType"])
 
 
+def get_concrete_artifact_map_from_globbed(upstreamArtifactPaths, artifactMap, strip_segments=["public/build", "public/logs"]):
+    """Given a list of artifacts from tasks (`upstreamArtifactPaths`) and a
+     mapping of names and/or patterns to destinations (`artifactMap`), return a
+     mapping that maps all of the specific artifacts from tasks int
+     destinations. Artifacts will have `strip_segments` removed from them before
+     being compared against `artifactMap` patterns.
```

Comment:
The strip_segments stuff seems a bit magic/surprising.  Could we avoid it, or pull it from the payload maybe?
Also AIUI strip_segments are removed from the eventual destination path, not just for comparison with the patterns?

Reply:
That's all correct - yes.

I don't have any objection to having it in the payload, seeing as this is not a translations-specific method. I think that providing a default if it's not there still might be sensible though (I imagine that most future use cases of this are likely to have the same prefixes on their artifacts). I could be convinced otherwise though.